src/main/scala/dpla/ingestion3/mappers/providers/TnMapping.scala

```diff
@@ -0,0 +1,265 @@
+      .flatMap(node => getByAttribute(node, "authority", "iso639-2b"))
+      .flatMap(extractStrings)
+      .map(nameOnlyConcept)
+
+  override def place(data: Document[NodeSeq]): Seq[DplaPlace] =
+  // <subject>
+  //  <geographic>
+  //    [text term] >> map to providedLabel
+  //  <cartographics>
+  //    <coordinates>
+  //      [coordinates] >> map to coordinates
+  (data \\ "mods" \ "subject")
+    .flatMap(place => {
+      extractStrings(place \ "geographic").map(Option(_))
+        .zipAll(extractStrings(place \ "cartographics" \ "coordinates").map(Option(_)), None, None)
+    })
+    .map( { case (name: Option[String], coord: Option[String]) => DplaPlace(name = name, coordinates = coord)} )
+
+
+  override def publisher(data: Document[NodeSeq]): Seq[EdmAgent] =
+  // <mods:originInfo><mods:publisher>
+    extractStrings(data \\ "mods" \ "originInfo" \ "publisher")
+      .map(nameOnlyAgent)
+
+  override def relation(data: Document[NodeSeq]): ZeroToMany[LiteralOrUri] = {
+    // <relatedItem>
+    //  <title>
+    //    <titlePart>[VALUE]
+    //  <location>
+    //    <url>[VALUE]
+    //
+    // when the type attribute DOES NOT equal "isReferencedBy" or "references" and
+    //  the "displayLabel" attribute DOES NOT equal "Project"
+    (data \\ "mods" \ "relatedItem")
+      .filterNot(node => (node \ "@type").text.equalsIgnoreCase("isReferencedBy"))
+      .filterNot(node => (node \ "@type").text.equalsIgnoreCase("references"))
+      .filterNot(node => (node \ "@displayLabel").text.equalsIgnoreCase("project"))
+      .flatMap(node =>
+        extractStrings(node \ "title" \ "titlePart") ++
+        extractStrings(node \ "location" \ "url"))
+      .map(eitherStringOrUri)
+  }
+
+  override def replacedBy(data: Document[NodeSeq]): ZeroToMany[String] =
+    // <relatedItem type="isReferencedBy">
+    //   <title>
+    //     <titlePart>[VALUE]
+    //   <location>
+    //     <url>[VALUE]
+    (data \\ "mods" \ "relatedItem")
+      .filter(node => (node \ "@type").text.equalsIgnoreCase("isReferencedBy"))
+      .flatMap(node =>
+        extractStrings(node \ "title" \ "titlePart") ++
+          extractStrings(node \ "location" \ "url"))
+
+  override def replaces(data: Document[NodeSeq]): ZeroToMany[String] =
+    // <relatedItem type="references">
+    //   <title>
+    //     <titlePart>[VALUE]
+    //   <location>
+    //     <url>[VALUE]
+    (data \\ "mods" \ "relatedItem")
+      .filter(node => (node \ "@type").text.equalsIgnoreCase("references"))
+      .flatMap(node =>
+        extractStrings(node \ "title" \ "titlePart") ++
+        extractStrings(node \ "location" \ "url"))
+
+  override def rights(data: Document[NodeSeq]): AtLeastOne[String] =
+    // <mods:accessCondition>
+    (data \\ "mods" \ "accessCondition")
+      .flatMap(node => getByAttribute(node, "type", "local rights statement")) // TODO should this filter be in place?
+      .flatMap(extractStrings)
```

Comment:
Calling out the `TODO` above in case you had intended to address it in this PR.

Reply:
That was for partner follow-up in CQA.